PathfinderAPI/Util/XML/ElementInfo.cs

```diff
@@ -28,60 +45,222 @@ public override string ToString()
-    {
-        foreach (var possibleInfo in list)
-        {
-            if (possibleInfo.Name == elementName)
-            {
-                return possibleInfo;
-            }
-        }
+    public static bool TryAsBoolean(this string content, out bool result)
+        => bool.TryParse(content, out result);
+    public static bool TryAsInt(this string content, out int result)
+        => int.TryParse(content, out result);
+    public static bool TryAsFloat(this string content, out float result)
+        => float.TryParse(content, out result);
 
-        return null;
+    public static bool AsBooleanSafe(this string content, bool defaultVal = default)
+        => content.TryAsBoolean(out var value)
+            ? value
+            : defaultVal;
+    public static int AsIntSafe(this string content, int defaultVal = default)
+        => content.TryAsInt(out var value)
+            ? value
+            : defaultVal;
+    public static float AsFloatSafe(this string content, float defaultVal = default)
+        => content.TryAsFloat(out var value)
+            ? value
+            : defaultVal;
+
+    public static bool AsBoolean(this string content, string valName = "content")
+        => content.TryAsBoolean(out var value)
+            ? value
+            : throw new FormatException($"Value of '{valName}' is '{content}' which is not true or false");
+    public static int AsInt(this string content, string valName = "content")
+        => content.TryAsInt(out var value)
+            ? value
+            : throw new FormatException($"Value of '{valName}' is '{content}' which is not an integer, e.g.: 0, 1, 2");
+    public static float AsFloat(this string content, string valName = "content")
+        => content.TryAsFloat(out var value)
+            ? value
+            : throw new FormatException($"Value of '{valName}' is '{content}' which is not a float, e.g.: 1.0");
+}
+
+public static class ElementInfoListExtensions
+{
+    public static ElementInfo GetElement(IEnumerable<ElementInfo> list, string elementName)
+    {
+        return list.FirstOrDefault(e => e.Name == elementName);
     }
-    public static bool TryGetElement(this List<ElementInfo> list, string elementName, out ElementInfo info)
+    public static bool TryGetElement(IEnumerable<ElementInfo> list, string elementName, out ElementInfo info)
     {
         info = GetElement(list, elementName);
         return info != null;
     }
+}
+
+[Obsolete("Use ElementInfoListExtensions")]
+public static class ListExtensions
+{
+    public static ElementInfo GetElement(this IEnumerable<ElementInfo> list, string elementName)
+        => ElementInfoListExtensions.GetElement(list, elementName);
+    public static bool TryGetElement(this IEnumerable<ElementInfo> list, string elementName, out ElementInfo info)
+        => ElementInfoListExtensions.TryGetElement(list, elementName, out info);
+}
+
+public static class ElementInfoDictionaryExtensions
```

Comment:
Are these not the same extension methods that already exist in PathfinderAPI.Utils.DictionaryExtensions?

Reply:
Either I forgot or it was done before we had those, this stuff was around for a long time before I committed it.